src/main/java/org/gaul/s3proxy/S3Exception.java

```diff
@@ -63,7 +65,12 @@ S3ErrorCode getError() {
 
     @Override
     public String getMessage() {
-        StringBuilder builder = new StringBuilder().append(error);
+        StringBuilder builder;
+        if (message != null) {
+            builder = new StringBuilder().append(message);
```

Comment:
Could you assign `StringBuilder` only once above and mutate it with `append(message)`, possibly with the ternary `?:` operator?

Reply:
I realized this simplifies a bit more: `message` must also be set and there is a corresponding `requireNonNull` check. Done.